3-reveal/migrations/5-IRS/4-Zambia-2020/deploy/zambia_irs_mop_up.psql

```diff
@@ -203,8 +234,10 @@ SELECT * FROM
             COALESCE(SUM(rooms_sprayed) FILTER (WHERE business_status IN ('Partially Sprayed', 'Complete')), 0) AS sprayed_rooms_sprayed,
             COALESCE(COUNT(structure_id) FILTER (WHERE duplicate = TRUE), 0) AS duplicates,
             COALESCE(COUNT(structure_id) FILTER (WHERE sprayed_duplicate = TRUE), 0) AS sprayed_duplicates,
-            array_concat_agg(notsprayed_reasons) FILTER (WHERE notsprayed_reasons <> '{}')  AS notsprayed_reasons
-        FROM zambia_irs_structures
+            array_concat_agg(notsprayed_reasons) FILTER (WHERE notsprayed_reasons <> '{}')  AS notsprayed_reasons,
+            (MAX(ARRAY[TO_JSON(event_date)#>>'{}', event_id]) FILTER (WHERE event_id is not null))[2] AS latest_spray_event_id,
```

Comment:
See: https://hakibenita.com/sql-group-by-first-last-value for why we did this way.  The TO_JSON magic is also the simplest way to get ISO-formatting.

Happy to consider other methods but it seems like we need to use arrays this way in postgres if we don't want to add additional join structure (I really don't).